sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -355,7 +377,14 @@ case class FileSourceScanExec(
   @transient
   private lazy val pushedDownFilters = {
     val supportNestedPredicatePushdown = DataSourceUtils.supportNestedPredicatePushdown(relation)
-    dataFilters.flatMap(DataSourceStrategy.translateFilter(_, supportNestedPredicatePushdown))
+    dataFilters
+      .filterNot(
+        _.references.exists {
+          case MetadataAttribute(_) => true
```

Comment:
How do we handle predicates against metadata columns? Do we ask Spark to run the filters?

I think it's better to push them down to the file source reader. For example, we can skip reading an entire file if the predicates have some check about the file name.

Reply:
Definitely a great point! That is something we want to do in follow-up PRs. We can probably only send files we want to scan (based on the filter), as you said.

Right now, we want to make sure of the correctness, see test `filter`.

Reply:
can we leave a TODO comment here?